scripts/deploy/cluster-up.sh

```diff
@@ -0,0 +1,300 @@
+
+  if [[ ! -f "$AZURE_CLUSTER_TEMPLATE_FILE" ]]; then
+    AZURE_CLUSTER_VALID_TEMPLATES=$(find "${AZURE_CLUSTER_TEMPLATE_ROOT}" -maxdepth 1 -printf "%P\n" | awk '{split($1,f,"."); printf (NR>1?", ":"") f[1]}')
+    echoerr "ERROR: The template '$AZURE_CLUSTER_TEMPLATE' is not known. Select one of the following values: $AZURE_CLUSTER_VALID_TEMPLATES."
+    printhelp
+    exit 1
+  fi
+
+  AZURE_CLUSTER_TEMPLATE_FILE=file://${AZURE_CLUSTER_TEMPLATE_FILE}
+else
+  AZURE_CLUSTER_TEMPLATE_FILE=${AZURE_CLUSTER_TEMPLATE}
+fi
+
+if [[ -z ${AZURE_CLUSTER_DNS_NAME:-} ]]; then
+  CLUSTER_PREFIX=$(whoami)
+  if [[ ${CLUSTER_PREFIX:-root} == "root" ]]; then
+    CLUSTER_PREFIX=k8s
+  fi
+  AZURE_CLUSTER_DNS_NAME=$(basename "$(mktemp -t "${CLUSTER_PREFIX}-${AZURE_CLUSTER_TEMPLATE}-${GIT_COMMIT}-XXX")")
+fi
+
+if [[ -z ${AZURE_RESOURCE_GROUP:-} ]]; then
+  AZURE_RESOURCE_GROUP=${AZURE_CLUSTER_DNS_NAME}-rg
+fi
+
+if [[ -z ${OUTPUT_DIR:-} ]]; then
+  OUTPUT_DIR="$GIT_ROOT/_output/$AZURE_CLUSTER_DNS_NAME"
+fi
+
+#
+# Install required tools
+#
+install_helm
+
+#
+# Create the Kubernetes cluster
+#
+echo "Creating cluster ${AZURE_CLUSTER_DNS_NAME}"
+"${GIT_ROOT}/scripts/deploy/azure-cluster-up.sh" \
+  --subscription "${AZURE_SUBSCRIPTION_ID}" \
+  --location "${AZURE_LOCATION}" \
+  --name "${AZURE_CLUSTER_DNS_NAME}" \
+  --output "${OUTPUT_DIR}" \
+  --k8s-version "${AZURE_K8S_VERSION}" \
+  --template "${AZURE_CLUSTER_TEMPLATE}"
+
+# Delete the cluster on subsequent errors.
+trap_push "\"${OUTPUT_DIR}/cluster-down.sh\"" err
+
+echo "Wait for cluster to become available..."
+# TODO Figure out a better way to determine cluster availability
+sleep 5m
+
+export KUBECONFIG="${OUTPUT_DIR}/kubeconfig/kubeconfig.${AZURE_LOCATION}.json"
+
+#
+# Install the Azure Disk CSI Driver
+#
+# echo "Installing Azure Disk CSI Driver..."
+# helm repo add azuredisk-csi-driver https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/master/charts
+# helm upgrade \
+#   --install \
+#   --namespace kube-system \
+#   azuredisk-csi-driver \
+#   azuredisk-csi-driver/azuredisk-csi-driver \
+#   -f "${AZURE_CLUSTER_TEMPLATE_ROOT}/${AZURE_CLUSTER_TEMPLATE}/azuredisk-csi-driver-values.yaml"
+# echo "Waiting for Azure Disk CSI Driver to start..."
+# kubectl wait pods --namespace kube-system --selector app.kubernetes.io/instance=azuredisk-csi-driver --for condition=ready --timeout=15m
+
+#
+# Apply label to Mayastor Node Candidates
+#
+echo "Labelling Mayastor Node Candidates..."
+kubectl label nodes --selector agentpool=agentpool openebs.io/engine=mayastor 
+
+#
+# Enable hugepages and prepare the nodes for reboot
+#
+echo "Enabling HugePages and marking all the nodes for reboot..."
+kubectl apply -f actions/hugepage-enabler-daemonset.yaml
+sleep 2m
+
+#
+# Restart nodes
+#
+echo "Restarting nodes..."
+kubectl apply -f actions/kured-config.yaml
+sleep 3m
+getNodesStatus
+echo "Nodes restart successful"
```

Comment:
What keeps the `kured` and `hugepage` DaemonSets from running again and rebooting the nodes over and over? I don't see where these DaemonSets are disabled or deleted.

Reply:
I have fixed this since I had faced this issue while testing. While push the updated changes along with other fixes.